src/H5Dchunk.c

```diff
@@ -1672,54 +1672,61 @@ H5D__create_chunk_file_map_all(H5D_chunk_map_t *fm, const H5D_io_info_t
 
         /* Allocate the file & memory chunk information */
-        if (NULL == (new_chunk_info = H5FL_MALLOC(H5D_chunk_info_t)))
-            HGOTO_ERROR(H5E_DATASET, H5E_CANTALLOC, FAIL, "can't allocate chunk info")
+        if (NULL == (new_piece_info = H5FL_MALLOC(H5D_piece_info_t)))
+            HGOTO_ERROR(H5E_DATASET, H5E_CANTALLOC, FAIL, "can't allocate piece info")
 
         /* Initialize the chunk information */
 
         /* Set the chunk index */
-        new_chunk_info->index = chunk_index;
-
-#ifdef H5_HAVE_PARALLEL
-        /* Store chunk selection information, for multi-chunk I/O */
-        if (io_info->using_mpi_vfd)
-            fm->select_chunk[chunk_index] = new_chunk_info;
-#endif /* H5_HAVE_PARALLEL */
+        new_piece_info->index = chunk_index;
 
         /* Set the file chunk dataspace */
-        if (NULL == (new_chunk_info->fspace = H5S_copy(tmp_fchunk, TRUE, FALSE)))
+        if (NULL == (new_piece_info->fspace = H5S_copy(tmp_fchunk, TRUE, FALSE)))
             HGOTO_ERROR(H5E_DATASPACE, H5E_CANTCOPY, FAIL, "unable to copy chunk dataspace")
-        new_chunk_info->fspace_shared = FALSE;
+        new_piece_info->fspace_shared = FALSE;
 
         /* If there are partial dimensions for this chunk, set the hyperslab for them */
         if (num_partial_dims > 0)
-            if (H5S_select_hyperslab(new_chunk_info->fspace, H5S_SELECT_SET, zeros, NULL, curr_partial_clip,
+            if (H5S_select_hyperslab(new_piece_info->fspace, H5S_SELECT_SET, zeros, NULL, curr_partial_clip,
                                      NULL) < 0)
                 HGOTO_ERROR(H5E_DATASET, H5E_CANTSELECT, FAIL, "can't create chunk selection")
 
         /* Set the memory chunk dataspace */
-        new_chunk_info->mspace        = NULL;
-        new_chunk_info->mspace_shared = FALSE;
+        new_piece_info->mspace        = NULL;
+        new_piece_info->mspace_shared = FALSE;
 
         /* Copy the chunk's scaled coordinates */
-        H5MM_memcpy(new_chunk_info->scaled, scaled, sizeof(hsize_t) * fm->f_ndims);
-        new_chunk_info->scaled[fm->f_ndims] = 0;
+        H5MM_memcpy(new_piece_info->scaled, scaled, sizeof(hsize_t) * fm->f_ndims);
+        new_piece_info->scaled[fm->f_ndims] = 0;
+
+        /* make connection to related dset info from this piece_info */
+        new_piece_info->dset_info = di;
 
         /* Insert the new chunk into the skip list */
-        if (H5SL_insert(fm->sel_chunks, new_chunk_info, &new_chunk_info->index) < 0) {
-            H5D__free_chunk_info(new_chunk_info, NULL, NULL);
+        if (H5SL_insert(fm->dset_sel_pieces, new_piece_info, &new_piece_info->index) < 0) {
+            H5D__free_piece_info(new_piece_info, NULL, NULL);
             HGOTO_ERROR(H5E_DATASPACE, H5E_CANTINSERT, FAIL, "can't insert chunk into skip list")
         } /* end if */
 
+        /* get chunk file address */
+        if (H5D__chunk_lookup(di->dset, new_piece_info->scaled, &udata) < 0)
```

Comment:
I'm guessing these chunk lookups have been moved from somewhere else? Just checking that we aren't adding extra metadata overhead.

Reply:
This call (and similar ones throughout the io_init functions) has been eliminted by commit [37cdba4](https://github.com/HDFGroup/hdf5/pull/2120/commits/37cdba43cde78c66b7ab1ffe68519311a96361a6)